packages/office-ui-fabric-react/etc/office-ui-fabric-react.api.md

```diff
@@ -8871,6 +8893,16 @@ export class Sticky extends BaseComponent<IStickyProps, IStickyState> {
     syncScroll: (container: HTMLElement) => void;
 }
 
+// @public (undocumented)
+export enum StickyContainerBehaviorType {
+    Default = 0,
+    StickyAlways = 2,
+    StickyOnScroll = 1

```

Comment:
Do we want to offer `StickyAlways` vs `StickyOnScroll`?

As a consumer wouldn't I always expect sticky elements to be "sticky" within a "StickyContainer"? 

I'm not sure I'd know which to choose.

Reply:
The default sticky behavior impacts PLT. It would calculate [distance from top for nonSticky div](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Sticky/Sticky.tsx#L249) even if user interaction has not started. If there is a large detailsList and detailsFooter is made sticky, this calculation helps to ensure detailsFooter is sticky and visible at the bottom without any user interaction (scrolling). If detailsFooter has 'StickyOnScroll' behavior, it would be visible only after first vertical scroll. For 'StickyOnScroll' calculations are started only after first vertical scroll. So 'StickyOnScroll' behaves same as 'Default' after first vertical scroll. In PLT, for 'StickyOnScroll' these calculations are skipped. 'StickyOnScroll' is most suitable for headers. Among all, 'StickyAlways' is the least expensive one. 'StickyAlways' ensures component is always sticky independent of vertical scroll and 'll always appear at it's sticky position. However, 'StickyAlways' is still different from css postion fixed. Let's say, there are multiple components which 've sticky behavior 'StickyAlways' and same sticky position (header/footer), 'StickyAlways' would not [need top or bottom values](https://codepen.io/anon/pen/QXJeGE) for these components. 
'StickyAlways' is most suitable if sticky position is 'Header' or 'Footer' and there is no content above or below the component respectively.

Reply:
The element 'll always be sticky within a StickyContainer. I am not sure if I understand your question correctly. There is some documentation:  packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.types.ts 
This behavior only provides a way to skip calculations.